src/azurehelper.ts

```diff
@@ -0,0 +1,166 @@
+import * as uuid from 'uuid';
+import { AzureEnvironment, DeviceTokenCredentials } from 'ms-rest-azure';
+import { ServiceClientCredentials } from 'ms-rest';
+import AuthorizationManagementClient = require('azure-arm-authorization');
+
+export interface IAzureHelper {
+    assignContributorRoleOnSubscription(principalId: string): Promise<boolean>;
+    assignContributorRoleOnResourceGroup(principalId: string, resourceGroupName: string): Promise<boolean>;
+    assignOwnerRoleOnSubscription(principalId: string): Promise<boolean>;
+    assignOwnerRoleOnResourceGroup(principalId: string, resourceGroupName: string): Promise<boolean>;
+    createRoleAssignmentWithRetry(principalId: string, roleId: string, scope: string): Promise<boolean>;
+    getAuthIssuserUrl(tenantId: string): string;
+    getStorageEndpointSuffix(): string;
+    getVMFQDNSuffix(): string;
+    getServiceBusEndpointSuffix(): string;
+    getPortalUrl(): string;
+    getCloudType(): string;
+}
+
+export class AzureHelper implements IAzureHelper {
+    private _environment: AzureEnvironment;
+    private _credentials: ServiceClientCredentials;
+    private _subscriptionId: string;
+    private MAX_RETRYCOUNT = 36;
+    private SLEEP_TIME = 5000;
+
+    constructor(environment: AzureEnvironment, subscriptionId: string, credentials: ServiceClientCredentials) {
+        this._environment = environment;
+        this._subscriptionId = subscriptionId;
+        this._credentials = credentials;
+    }
+
+    public assignContributorRoleOnSubscription(principalId: string): Promise<boolean> {
+        const roleId = 'b24988ac-6180-42a0-ab88-20f7382dd24c';
+        const scope = `/subscriptions/${this._subscriptionId}`;
+        return this.createRoleAssignmentWithRetry(principalId, scope, roleId);
+    }
+
+    public assignContributorRoleOnResourceGroup(principalId: string, resourceGroupName: string): Promise<boolean> {
+        const roleId = 'b24988ac-6180-42a0-ab88-20f7382dd24c';
```

Comment:
Add a comment that this GUID represents roleId as `Contributor` in AAD. Also make this a static variable and call it CONTRIBUTOR_ID